tests/templates/test_layers/test_basic_entangler.py

```diff
@@ -150,12 +143,46 @@ def test_shape(self, n_layers, n_wires, expected_shape):
         assert shape == expected_shape
 
 
-class TestGradients:
-    """Tests that the gradient is computed correctly in all interfaces."""
+def circuit_template(weights):
+    qml.templates.BasicEntanglerLayers(weights, range(3))
+    return qml.expval(qml.PauliZ(0))
+
+
+def circuit_decomposed(weights):
+    qml.RX(weights[0, 0], wires=0)
+    qml.RX(weights[0, 1], wires=1)
+    qml.RX(weights[0, 2], wires=2)
+    qml.CNOT(wires=[0, 1])
+    qml.CNOT(wires=[1, 2])
+    qml.CNOT(wires=[2, 0])
+    return qml.expval(qml.PauliZ(0))
+
+
+class TestInterfaces:
+    """Tests that the template is compatible with all interfaces, including the computation
+    of gradients."""
+
+    def test_list_and_tuples(self, tol):
+        """Tests common iterables as inputs."""
+
+        weights = [[0.1, -1.1, 0.2]]
+
+        dev = qml.device("default.qubit", wires=3)
+
+        circuit = qml.QNode(circuit_template, dev)
+        circuit2 = qml.QNode(circuit_decomposed, dev)
+
+        res = circuit(weights)
+        res2 = circuit2(weights)
+        assert qml.math.allclose(res, res2, atol=tol, rtol=0)
```

Comment:
```suggestion
        np.testing.assert_allclose(res, res2, atol=tol, rtol=0)
```

This works with TF/Torch/JAX tensor-types too!

Reply:
I just checked and works here, but not when a tensor has a grad attached...so will not change it everywhere...